sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt2Tests.java

```diff
@@ -0,0 +1,335 @@
+                        "             (-2.9::numeric)," +
+                        "             (-2.5::numeric)," +
+                        "             (-1.5::numeric)," +
+                        "             (-0.5::numeric)," +
+                        "             (0.0::numeric)," +
+                        "             (0.5::numeric)," +
+                        "             (1.5::numeric)," +
+                        "             (2.5::numeric)) t(x);\n" +
+                        "  x   | int2_value \n" +
+                        "------+------------\n" +
+                        " -2.9 |         -2\n" +
+                        " -2.5 |         -2\n" +
+                        " -1.5 |         -1\n" +
+                        " -0.5 |         -0\n" +
+                        "  0.0 |          0\n" +
+                        "  0.5 |          0\n" +
+                        "  1.5 |          1\n" +
+                        "  2.5 |          2"
+        );
+    }
+
+    // Ignored because this fails in Postgres but here we get:
+    @Test @Ignore("Integer wrapping: https://github.com/feldera/feldera/issues/1186")
+    public void testSelectOverflow() {
+        String error_message = "INT2 out of range";
+        // We get:
+        // L: (Some(-32767), Some(2))x1 --> wraps around
+        // L: (Some(-1234), Some(-2468))x1
+        // L: (Some(0), Some(0))x1
+        // L: (Some(1234), Some(2468))x1
+        // L: (Some(32767), Some(-2))x1 --> wraps around
+        // Taken from: https://github.com/postgres/postgres/blob/c161ab74f76af8e0f3c6b349438525ad9575683b/src/test/regress/expected/int2.out#L202C1-L203C30
+        this.shouldFail("SELECT i.f1, i.f1 * 2::INT2 AS x FROM INT2_TBL i", error_message);
+
+        // We get:
+        // L: (Some(-32767), Some(-32765))x1
+        // L: (Some(-1234), Some(-1232))x1
+        // L: (Some(0), Some(2))x1
+        // L: (Some(1234), Some(1236))x1
+        // L: (Some(32767), Some(-32767))x1 --> wraps around
+        // https://github.com/postgres/postgres/blob/c161ab74f76af8e0f3c6b349438525ad9575683b/src/test/regress/expected/int2.out#L223
+        this.shouldFail( "SELECT i.f1, i.f1 + '2'::INT2 AS x FROM INT2_TBL i", error_message);
+
+        // Similarly,
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT2 AS x FROM INT2_TBL i;", error_message);
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    // error [E0080]:
+    //  |
+    //2 | println!("{}", -32768i16 % -1i16);
+    //  |                ^^^^^^^^^^^^^^^^^ attempt to compute `i16::MIN % -1_i16`, which would overflow
+    //  |
+    //  = note: `#[deny(unconditional_panic)]` on by default
+    //  |
+    //2 | println!("{}", -32768i16 % -1i16);
+    //  |                ^^^^^^^^^^^^^^^^^ overflow in signed remainder (dividing MIN by -1)
+    @Test @Ignore("Modulo edge case integer overflow: https://github.com/feldera/feldera/issues/1187")
+    public void testINT2MINOverflow() {
```

Comment:
This test should use the new `runtimeFail` function from #1183

Reply:
This comment was wrong. I think that the Rust implementation has a bug here. I will resolve this conversation.

Reply:
The actual issue mentioned here should be #1195